ts/features/pn/components/PnMessageDetails.tsx

```diff
@@ -67,21 +63,14 @@ export const PnMessageDetails = (props: Props) => {
   const currentFiscalCode = useIOSelector(profileFiscalCodeSelector);
   const frontendUrl = useIOSelector(PnConfigSelector).frontend_url;
 
-  const maybePayment = props.message.recipients.find(
-    _ => _.taxId === currentFiscalCode
-  )?.payment;
-
-  const noticeNumber = PaymentNoticeNumber.decode(maybePayment?.noticeCode);
-  const creditorTaxId = OrganizationFiscalCode.decode(
-    maybePayment?.creditorTaxId
+  const maybePayment = pipe(
+    props.message.recipients,
+    AR.findFirst(_ => _.taxId === currentFiscalCode),
+    O.chainNullableK(_ => _.payment),
+    O.map(identity),
```

Comment:
This `O.map(identity),` step should not be necessary. Can we remove it?

Reply:
changed https://github.com/pagopa/io-app/pull/4624/commits/3ce6c4cdb06a1bf3cfa6e630fc0322bf75a6fdee